### PR TITLE
Making logging configurable

### PR DESCRIPTION
**Pull request by mohsen0**

### CODE REVIEW

**Comment by mohsen0** [comment]
It seems something is wrong with CI setup, can you take a look?
```
The command "terraform fmt -check=true" exited with 0.
0.64s$ terraform validate -var "region=${AWS_REGION}" -var "subnets=[]" -var "vpc_id=vpc-abcde012" -var "load_balancer_name=my-lb" -var "log_bucket_name=my-log-bucket" -var "security_groups=[]"
Error: provider.aws: "region": required field is not set
The command "terraform validate -var "region=${AWS_REGION}" -var "subnets=[]" -var "vpc_id=vpc-abcde012" -var "load_balancer_name=my-lb" -var "log_bucket_name=my-log-bucket" -var "security_groups=[]"" exited with 1.
```

**Comment by mohsen0** [comment]
Hi @antonbabenko Can this be reviewed? sorry to bother you but I want to see if it is possible to get this faster since it is such a small thing

**Comment by antonbabenko** [comment]
Hi @mohsen0 ! Yes, I can review, but I am not feeling comfortable with everything that is going after (eg, automated tests, release).

@brandoconnor usually reviews such PRs pretty fast.

File: variables.tf [hunk 1]
```diff
@@ -80,8 +80,14 @@ variable "load_balancer_update_timeout" {
   default     = "10m"
 }
 
+variable "log_enable" {
+  description = "Enable logging"
+  default     = false
```
Comment (antonbabenko) [inline variables.tf]: I would prefer to set the default to `true` to keep backward compatiblity for people who are using `master` branch of this module.

Comment (mohsen0) [reply variables.tf]: ok, changed

**Comment by mohsen0** [comment]
Hi @brandoconnor , can you review this work?
Thanks
Mohsen

**Comment by brandoconnor** [comment]
@mohsen0 - I've had a busy week but should be able to release this by end of day.

I think the change looks fine. I wonder if there are any combinations of `log_bucket_name`, `log_location_prefix`, and `log_enable` that won't work together... If so, those should be mentioned in the README. What have you tried?

**Comment by brandoconnor** [comment]
RE: CI - I don't know how to allow PRs to initiate CI using the secret environment variables provided to Travis. There are some docs on it here:
https://docs.travis-ci.com/user/pull-requests/
https://docs.travis-ci.com/user/environment-variables/

None of that seems to indicate there's a way to allow PRs to use environmnet variables that are secrets but maybe I've misread. That's the reason why CI is failing.

**Comment by brandoconnor** [comment]
I'm going to make a change or two once this is merged but pulling this in now. Thanks for the contribution @mohsen0

**Comment by brandoconnor** [comment]
Ugh and the problem was not too far from my comment. If you try to specify `false` for this variable but don't provide a valid bucket for `log_bucket_name`, apply fails. This leaves the user needing to specify a valid bucket, or even more awkward, creating one within the module just to satisfy this requirement.

@mohsen0 did you test any of that before submitting? I can think of different ways to make this work happen but it's somewhat clunky. I won't be cutting this in a release until the problem is resolved.

**Comment by vas1468** [comment]
Same issue here when I am trying to use 
`log_enable          = "${var.log_enable}"` within the module.

After running terraform plan getting the error:

`Initializing modules...
- module.alb
  Found version 3.2.0 of terraform-aws-modules/alb/aws on registry.terraform.io
  Getting source "terraform-aws-modules/alb/aws"

Initializing the backend...

Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.

Error: module "alb": "log_enable" is not a valid argument`

**Comment by brandoconnor** [comment]
It's a shame to see contributors not taking ownership of their changes. I'll be rolling this back now.

**Comment by mohsen0** [comment]
I reverted back my changes, I did spend time on this, it will be complicated to keep the current default variables the same thing and make this change work.  
There is another way to make this configurable and having two```resource "aws_lb" "application"``` with and without access_logs section and using count condition which one to be created.

I can make that change if you are happy with that :), 
anyhow I will raise another PR with the revert so you merge that for now.

**Comment by mohsen0** [comment]
https://github.com/terraform-aws-modules/terraform-aws-alb/pull/62

**Comment by brandoconnor** [comment]
Thank you. Accepting that PR now.

Re: the count on the two `aws_lb` resources - I'm fine with going this route. It adds some complexity that might spiral out of control since other resources have explicit dependencies on that resource and outputs must also be juggled with a lot of interpolation. If it works, I'd take it because as an end user, it's just an implementation detail that you shouldn't need to worry about. That's the whole point, right? Abstract away the details.

**Pull request merged**